### PR TITLE
Update to `4.0.0`

### LINKED ISSUE #26: Add response specific types
As of right now, all endpoints return a generic dynamic `ComposableRequest.Response`.\
Going forward I'm going to implement some `protocol`s for the most common responses.

e.g.
```swift
public protocol User
public extension User where Self: DynamicResponse {
     var username: String? { return self["username"].string() }
}
```

### LINKED ISSUE #56: Move `Storage` to ComposableRequest
Most of client-related code has been refactored into [**ComposableRequest**](https://github.com/sbertix/ComposableRequest), but there are still some oddities, like `Storage`, and its concrete implementation.

This requires moving `KeychainSwift` to **ComposableRequest** (most likely into a similar **ComposableRequestCrypto**), but other than that, it should be pretty quick to implement.

### LINKED ISSUE #57: Additional endpoints
- [x] I've searched past issues and I couldn't find reference to this. 

**Is your feature request related to a problem? Please describe.**
I wanted to implement some features in my project but some endpoints are missing. For example, you can retrieve posts' comments, but can't comment a post. You can retrieve saved posts, but can't save a post.

**Describe the solution you'd like**
Provide new endpoints:
- `Endpoint.Media.Posts.comment(for identifier: String, content: String)` where identifier is the post identifier (like when retrieving comments) and content is the content of the new comment.
- `Endpoint.Feed.save(for identifier: String)` to save a post with its identifier
- same for stories, to upload a story, with eventually a tag or a mention

### PR DESCRIPTION
**Pull request by sbertix**
# 4.0.0
_Renamed to_ ~3.2.0~ **4.0.0** _because of breaking changes._

## Enchancements
- [x] Move `Storage` into `ComposableRequest` (Closes #56)
- [x] Add new `Direct` `Endpoint`s
   - [x] Pending inbox
   - [x] Queried recipients
   - [x] Presence
- [x] Add new `Feed` `Endpoint`s
   - [x] Timeline
- [x] Add new `Friendship` `Endpoint`s
   - [x] Multiple checks at the same type
- [x] Add new `Location` `Endpoint`s
   - [x] Search places
   - [x] Place summary
   - [x] Place stories
- [x] Add new `Media.Posts` `Endpoint`s (Closes #57, #55)
   - [x] Upload pictures (**SwiftagramCrypto**)
   - [ ] ~Upload videos (**SwiftagramCrypto**)~ → _moved to a future release_
   - [ ] ~Upload stories (**SwiftagramCrypto**)~ → _we need more time to actually decide how to deal with all available configurations (an `enum`, perhaps?)_
   - [x] Delete posts (**SwiftagramCrypto**)
   - [x] Comment on posts (**SwiftagramCrypto**)
   - [x] Delete comments on posts (**SwiftagramCrypto**)
   - [x] Like and unlike comments
   - [x] Save and unsave posts
   - [x] Archive and unarchive posts (**SwiftagramCrypto**)
- [x] Create `Deprecations.swift` in both **Swiftagram** and **SwiftagramCrypto** with soon to be removed declarations
- [x] Deprecate `Endpoint.Feed`
- [x] Migrate selected `Endpoint`s to new `Wrappable` return types (Closes #26)
   - [x] **Swiftagram**
      - [x] `Archive`
      - [x] `Direct`
      - [x] `Friendship`
      - [x] `Highlights`
      - [x] `Location`
      - [x] `Media`
      - [x] `User`
   - [x] **SwiftagramCrypto**

## Misc
- [x] Move **ComposableRequest** dependency from `development` to _next minor_ after its release
- [x] Update README.md
- [x] Add tests for all new functionalities
- [x] Update _Examples/Followers_
- [x] Move **Swiftagram** dependency in _Examples/Followers_ from `development` to _next minor_ before final release

### CODE REVIEW

File: Sources/SwiftagramCrypto/Endpoints/EndpointMedia.swift [hunk 1]
```diff
@@ -37,13 +60,267 @@ public extension Endpoint.Media {
...
+    /// - parameters:
+    ///     - image: A `NSImage` representation of an image.
+    ///     - caption: An optional `String` holding the post's caption.
+    ///     - users: An optional collection of `UserTag`s. Defaults to `nil`.
+    ///     - location: An optional `Location`. Defaults to `nil`.
+    static func upload<U: Collection>(image: NSImage,
+                                      captioned caption: String?,
+                                      tagging users: U? = nil,
+                                      at location: Location? = nil) -> Endpoint.DisposableResponse where U.Element == UserTag {
+        guard let data = image.tiffRepresentation else { fatalError("Invalid `UIImage`.") }
```
Comment (anonrig) [inline Sources/SwiftagramCrypto/Endpoints/EndpointMedia.swift:209]: tiffRepresentation is not available in Mac Catalyst, and therefore build fails.

**Comment by sbertix** [comment]
I've scrapped `Response` and `ResponseMappable`, and replaced them with `Wrapper` (completely rewritten) and `Wrapped` (greatly improved), as I don't want to change these anymore in the future.\
All endpoints in `Tests` seem to be working fine, but since picture uploading is not one of them, I'm gonna need to investigate further tomorrow.\
If you happen to actually try it before that, please send me a message on Telegram on whether it's working or not @NathanFallet 💪

**Comment by sbertix** [comment]
For anyone watching this PR,I'm planning to release `4.0.0` by the end of the week.\
Apologies for the delays.

**Pull request merged**